File: Delete-Columns-to-Make-Sorted.ts
1function minDeletionSize(strs: string[]): number {
2    const m = strs.length;
3    const n = strs[0].length;
4    let ans = 0;
5
6    for (let j = 0; j < n; j++) {
7        for (let i = 1; i < m; i++) {
8            if (strs[i][j] < strs[i - 1][j]) {
9                ans++;
10                break;
11            }
12        }
13    }
14
15    return ans;
16}
17

File: DCP-12-25/Delete-Columns-to-Make-Sorted.py
1from typing import List
2
3class Solution:
4    def minDeletionSize(self, strs: List[str]) -> int:
5        count = 0
6        rows = len(strs)
7        cols = len(strs[0])
8
9        for c in range(cols):
10            for r in range(1, rows):
11                if strs[r][c] < strs[r - 1][c]:
12                    count += 1
13                    break
14        return count